classes/shop/Shop.php

```diff
@@ -660,6 +660,11 @@ public static function addTableAssociation($table_name, $table_details)
         return true;
     }
 
+    public static function removeTableAssociation(string $tableName): void
```

Comment:
This one needs to be discussed.
So following test failed before 
`./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s cart --tags cart-to-order`.
What happened is that once I added cart_rule to $asso_tables its been there whole test suite.
So when the moment comes to create order cart rules in PaymentModule::1206, the new voucher is added and already assigned to cart_rule_shop. Later on payment module performs CartRule::copyConditions which then tries to add duplicate cart_rule_shop. So just to be safe, after adding/editing cart rule in "new" way, I restore previous $asso_tables, so in case its added/edited in old way again, it doesn't bring unexpected results.

Reply:
I'm not a fan of this "hack" Either we fully handle CartRule shop association like other multi-shop entities or we don't But here it's a mix of both strategies on a feature that is already way too complex and unstable.

I would prefer to correctly handle this fully like the other Entities, even if it means cleaning the legacy a bit But that implies once we do this we don't have a choice but to go through the whole migration process of this page no matter what and it's blocking for the 9.0 release

I'd like @MatShir opinion on this, can we consider this page's migration as blocking? And invest enough people working on it to make sure we completely finish it before feature freeze?